### ISSUE TITLE
Color Handling in Background Rendering

### ISSUE THREAD

**Issue by DartPads** [espresso3389/pdfrx#286]
I’m using pdfrx ^1.0.97 to render PDF pages into images (JPEG or PNG) in a background isolate. While the rendering works fine, I’ve noticed that the colors in the resulting images are not matching what I see when viewing the PDF in a typical viewer (Acrobat/Preview). 

Previously, when I used pdf_render and Flutter’s ui.Image on the main isolate, the colors seemed more faithful, likely due to Flutter’s sRGB color pipeline. However, I need to render in a background isolate now, and that pipeline isn’t available.

What I’m Trying to Achieve

1. Preserve (or closely match) the original PDF’s colors, especially when the PDF uses specific ICC profiles or CMYK color spaces.
2. Run the rendering in a background isolate (without using Flutter’s ui.Image.createImage()).

Code Example

Here’s a simplified version of my background isolate code:

```
import 'dart:typed_data';
import 'dart:isolate';
import 'package:pdfrx/pdfrx.dart' as rx;
import 'package:image/image.dart' as img;

void renderPdfPageInIsolate(Map<String, dynamic> params) async {
  final String filePath = params['filePath'];
  final SendPort sendPort = params['sendPort'];

  try {
    final rx.PdfDocument doc = await rx.PdfDocument.openFile(filePath);
    final rx.PdfPage page = doc.pages[0];

    // Render at 300 DPI, RGBA format:
    final rx.PdfPageImage pageImage = await page.render(
      width: (page.width * (300 / 72)).ceil(),
      height: (page.height * (300 / 72)).ceil(),
      format: rx.PdfPageFormat.RGBA8,
    );

    // Convert to an image.dart Image
    final Uint8List rgbaBytes = pageImage.pixels;
    final img.Image decodedImage = img.Image.fromBytes(
      width: pageImage.width,
      height: pageImage.height,
      bytes: rgbaBytes.buffer,
      // RGBA data
      numChannels: 4,
      format: img.Format.uint8,
    );

    // Encode to JPG:
    final Uint8List jpgBytes = img.encodeJpg(decodedImage, quality: 80);

    // ... Save the image, etc.
    sendPort.send("Rendered PDF Page!");
  } catch (e) {
    sendPort.send("Error: $e");
  }
}
```

Do you recommend any approach to replicate Flutter’s default sRGB color handling while staying in a background isolate? Any guidance or workarounds would be greatly appreciated!

Thank you for your hard work on pdfrx—I love the plugin and hope to see it continue to evolve!

**Comment by espresso3389** [comment]
Pdfrx, specifically in the case, pdfium, just render images in sRGB. And, I believe image package also does nothing for color management though `fromBytes` method accepts an ICC color profile. So I expect the output is pure sRGB image.

So, only the diffrence is the engine; pdfium vs. Apple Quartz(Preview)/Acrobat.

If you do the JPEG output process on the foreground UI isolate, does the color match to ones by Preview/Acrobat?

**Comment by espresso3389** [comment]
Oh, I want to explain my intention. I suspect that the JPEG outputs are identical regardless of UI/background isolates.

**Comment by DartPads** [comment]
> If you do the JPEG output process on the foreground UI isolate, does the color match to ones by Preview/Acrobat?

Yes. Please see the jpeg below which was converted from the following [PDF](https://www.bseindia.com/xml-data/corpfiling/AttachHis/9a5fc701-2db0-49a5-82d7-94bcd95378d0.pdf)

![GgGemN6bYAcIYpO](https://github.com/user-attachments/assets/d1e83b33-d99b-422f-8b1d-be877df54630)

**Comment by espresso3389** [comment]
It seems to be too late. But the problem is very easy, the channel order; RGBA and BGRA.
